patches/server/1044-Add-Llama-caravan-events.patch

```diff
@@ -0,0 +1,98 @@
+@@ -437,8 +437,19 @@ public class Llama extends AbstractChestedHorse implements VariantHolder<Llama.V
+         }
+     }
+ 
++    // Paper start - Add Llama caravan events
+     public void leaveCaravan() {
++        leaveCaravan(io.papermc.paper.event.entity.LlamaLeaveCaravanEvent.Reason.UNKNOWN);
++    }
++    // Paper end - Add Llama caravan events
++
++    public void leaveCaravan(io.papermc.paper.event.entity.LlamaLeaveCaravanEvent.Reason reason) { // Paper - add reason
+         if (this.caravanHead != null) {
++            // Paper start - Add Llama caravan events
++            if (!new io.papermc.paper.event.entity.LlamaLeaveCaravanEvent(getBukkitEntity(), reason).callEvent()) {
++                return;
++            }
++            // Paper end - Add Llama caravan events
+             this.caravanHead.caravanTail = null;
+         }
+ 
+@@ -446,10 +457,26 @@ public class Llama extends AbstractChestedHorse implements VariantHolder<Llama.V
+     }
+ 
+     public void joinCaravan(Llama llama) {
++        joinCaravan(llama, io.papermc.paper.event.entity.LlamaJoinCaravanEvent.Reason.UNKNOWN);
++    }
++
++    public void joinCaravan(Llama llama, io.papermc.paper.event.entity.LlamaJoinCaravanEvent.Reason reason) { // Paper - add reason
++        // Paper start - Add Llama caravan events
++        if (!new io.papermc.paper.event.entity.LlamaJoinCaravanEvent(getBukkitEntity(), llama.getBukkitEntity(), reason).callEvent()) {
++            return;
++        }
++        // Paper end - Add Llama caravan events
+         this.caravanHead = llama;
+         this.caravanHead.caravanTail = this;
+     }
+ 
++    // Paper start
++    @Override
++    public org.bukkit.craftbukkit.entity.CraftLlama getBukkitEntity() {
++        return (org.bukkit.craftbukkit.entity.CraftLlama) super.getBukkitEntity();
++    }
++    // Paper end
++
+     public boolean hasCaravanTail() {
+         return this.caravanTail != null;
+     }
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftLlama.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftLlama.java
+index 0ad16ee7b33582d214dab41eeee378d52c8e38ed..a7b1e37232dee1b1863f4f68c252b21fe69fa471 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftLlama.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftLlama.java
+@@ -67,12 +67,12 @@ public class CraftLlama extends CraftChestedHorse implements Llama, com.destroys
+ 
+     @Override
+     public void joinCaravan(@org.jetbrains.annotations.NotNull Llama llama) {
+-        this.getHandle().joinCaravan(((CraftLlama) llama).getHandle());
++        this.getHandle().joinCaravan(((CraftLlama) llama).getHandle(), io.papermc.paper.event.entity.LlamaJoinCaravanEvent.Reason.PLUGIN);
+     }
+ 
+     @Override
+     public void leaveCaravan() {
+-        this.getHandle().leaveCaravan();
++        this.getHandle().leaveCaravan(io.papermc.paper.event.entity.LlamaLeaveCaravanEvent.Reason.PLUGIN);
```

Comment:
Despite this being inside a paper block, best to add an inline paper comment with a comment of what is changing.